--- php-5.6.9/ext/pdo_dblib/dblib_stmt.c	2015-05-14 02:13:33.000000000 +0300
+++ php-src/ext/pdo_dblib/dblib_stmt.c	2015-07-03 10:27:20.791384633 +0300
@@ -203,8 +203,15 @@
 	}
 	
 	struct pdo_column_data *col = &stmt->columns[colno];
-	
-	col->name = (char*)dbcolname(H->link, colno+1);
+
+	char *fname = (char*)dbcolname(H->link, colno+1);
+	char computed_buf[16];
+	if (fname && *fname) {
+		col->name = estrdup(fname);
+	} else {
+		snprintf(computed_buf,16,"computed%d", colno);
+		col->name = estrdup(computed_buf);
+	}
 	col->maxlen = dbcollen(H->link, colno+1);
 	col->namelen = strlen(col->name);
 	col->param_type = PDO_PARAM_STR;
